lib/Data/Validate/Sanctions.pm

```diff
@@ -123,24 +127,33 @@ sub get_sanctioned_info {    ## no critic (RequireArgUnpacking)
             $matched_file = $file;
 
             # Some clients in sanction list can have more than one date of birth
-            # Comparison is made using the epoch value
-            my $client_dob_epoch   = Date::Utility->new($date_of_birth)->epoch;
-            my $sanctions_dob_list = $data->{$file}->{names_list}->{$sanctioned_name}->{dob_epoch};
+            # Comparison is made using the epoch and year values
+            my $client_dob_date  = Date::Utility->new($date_of_birth);
+            my $client_dob_epoch = $client_dob_date->epoch;
+            my $client_dob_year  = $client_dob_date->year;
 
-            # If the dob_epoch is missing from the sanctions.yml, automatically mark
+            my $sanctions_epoch_list = $data->{$file}->{names_list}->{$sanctioned_name}->{dob_epoch} // [];
+            my $sanctions_year_list  = $data->{$file}->{names_list}->{$sanctioned_name}->{dob_year}  // [];
+            $dob_other = $data->{$file}->{names_list}->{$sanctioned_name}->{dob_other} // [];
+
+            # If the dob_epoch and dob_year are missing from the sanctions.yml, automatically mark
             # the client as a terrorist, regardless of further checks
-            unless (@$sanctions_dob_list) {
+            unless (@$sanctions_epoch_list or @$sanctions_year_list) {
                 $dob_missing = 1;
             }
 
-            $checked_dob = any { $_ eq $client_dob_epoch } @{$sanctions_dob_list};
-
+            $checked_dob = any { $_ eq $client_dob_epoch } @{$sanctions_epoch_list};
             return _possible_match($matched_file, $matched_name, 'Date of birth matches', $date_of_birth) if $checked_dob;
+
+            $checked_dob = any { $_ eq $client_dob_year } @{$sanctions_year_list} unless $checked_dob;
+            return _possible_match($matched_file, $matched_name, 'Year of birth matches', $client_dob_year) if $checked_dob;
         }
     }
-
+    
+    my $reason = 'Name is similar';
+    $reason = 'Name is similar - dob info: ' . join (',', @$dob_other) if scalar @$dob_other;
     # Return a possible match if the name matches and no date of birth is present in sanctions
-    return _possible_match($matched_file, $matched_name, 'Name is similar', 'N/A') if ($matched_name && $dob_missing);
+    return _possible_match($matched_file, $matched_name, $reason, 'N/A') if $matched_name && $dob_missing;
```

Comment:
I may be mistaken, but  `$dob_missing` and `@$dob_other` have a similar meaning. Could it be that we have @$dob_other and matched date?

Reply:
$dob_other was a redundant variable. I removed the redundant variable, renamed $dob_missing and polished the logic a little bit, specially here: https://github.com/binary-com/perl-Data-Validate-Sanctions/pull/32/files#diff-7e74c4e3dc4bc7a8c2b436610581ade9R135-R138